pegasus/helpers/email_preference_helpers.rb

```diff
@@ -0,0 +1,47 @@
+require 'cdo/email_preference_constants'
+require 'cdo/email_validator'
+
+class EmailPreferenceHelper
+  def self.upsert(email:, opt_in:, ip_address:, source:, form_kind:)
```

Comment:
Similar to above, I'd prefer this be called `upsert!` (with the bang) to be more consistent with ActiveRecord, since it raises errors on failure.